substrate/utils/frame/benchmarking-cli/src/extrinsic/bench.rs

```diff
@@ -84,29 +85,34 @@ where
 		params: BenchmarkParams,
 		inherent_data: sp_inherents::InherentData,
 		digest_items: Vec<DigestItem>,
+		record_proof: bool,
 	) -> Self {
-		Self { client, params, inherent_data, digest_items, _p: PhantomData }
+		Self { client, params, inherent_data, digest_items, record_proof, _p: PhantomData }
 	}
 
 	/// Benchmark a block with only inherents.
-	pub fn bench_block(&self) -> Result<Stats> {
-		let (block, _) = self.build_block(None)?;
+	///
+	/// Returns the Ref time stats and the proof size.
+	pub fn bench_block(&self) -> Result<(Stats, u64)> {
+		let (block, _, proof_size) = self.build_block(None)?;
 		let record = self.measure_block(&block)?;
-		Stats::new(&record)
+
+		Ok((Stats::new(&record)?, proof_size))
 	}
 
 	/// Benchmark the time of an extrinsic in a full block.
 	///
 	/// First benchmarks an empty block, analogous to `bench_block` and use it as baseline.
 	/// Then benchmarks a full block built with the given `ext_builder` and subtracts the baseline
 	/// from the result.
-	/// This is necessary to account for the time the inherents use.
-	pub fn bench_extrinsic(&self, ext_builder: &dyn ExtrinsicBuilder) -> Result<Stats> {
-		let (block, _) = self.build_block(None)?;
+	/// This is necessary to account for the time the inherents use. Returns ref time stats and the
+	/// proof size.
+	pub fn bench_extrinsic(&self, ext_builder: &dyn ExtrinsicBuilder) -> Result<(Stats, u64)> {
+		let (block, _, base_proof_size) = self.build_block(None)?;
```

Comment:
Why do we use the proof size for build block and not for `execute_block` done in `measure_block`.

I would expect operations such as `api.inherent_extrinsics` in `create_inherents` and maybe some other build specific operation to not be needed in the proof.

Reply:
Because this logic should be the same as what we use in cumulus to compute the proof to send to the relay chain. And there we also use the block building proof.

We could maybe change this, since in the PVF we only call `execute_block`. But the author of the block does not re-execute the block after building, so we only have the build-proof available anyway.